src/gt4py/analysis/passes.py

```diff
@@ -1194,38 +1194,37 @@ class DemoteLocalTemporariesToVariablesPass(TransformPass):
     """
 
     class CollectDemotableSymbols(gt_ir.IRNodeVisitor):
-        def __init__(self):
-            pass
-
         @classmethod
         def apply(cls, node: gt_ir.StencilImplementation) -> Set[str]:
             collector = cls()
             return collector(node)
 
         def __call__(self, node: gt_ir.StencilImplementation) -> Set[str]:
             assert isinstance(node, gt_ir.StencilImplementation)
-            self.demotables = set(node.temporary_fields)
-            self.seen_symbols: Set[str] = set()
-            self.stage_symbols: Optional[Set[str]] = None
+            self.demotables: Dict[str, str] = {
+                field_name: "" for field_name in node.temporary_fields
+            }
```

Comment:
Can you add a docstring below this that describes what `self.demotables` stores?